src/Microsoft.AspNet.OData.Shared/Routing/Conventions/RequestMethodExtensions.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNet.OData.Interfaces;
+
+namespace Microsoft.AspNet.OData.Routing.Conventions
+{
+    internal static class RequestMethodExtensions 
+    {
+        internal static ODataRequestMethod GetRequestMethodOrPreflightMethod(this IWebApiRequestMessage request)
+        {
+            if (request.Method == ODataRequestMethod.Options)
+            {
+                IEnumerable<string> values;
+                request.Headers.TryGetValues("Access-Control-Request-Method", out values);
```

Comment:
When it returns false I do not check the header values and I resolve the method to the same method in case the user has added their own way to resolve options requests then this would go to that. 
Also according to the cors specification the method should be populated with one of the known request methods.

Reply:
I see john more explicitly addressed my concern here